File: src/main/java/com/example/financetracker/service/CategoryService.java
package com.example.financetracker.service;

import com.example.financetracker.dtos.request.CategoryRequest;
import com.example.financetracker.dtos.response.CategoryResponse;

import java.util.List;

public interface CategoryService {
    CategoryResponse createCategory(CategoryRequest request);
    List<CategoryResponse> getAllCategory();
    CategoryResponse getCategoryById(Long id);
    CategoryResponse updateCategory(Long id, CategoryRequest request);
    void deleteCategory(Long id);
}

File: finance-tracker-frontend/src/services/categoryService.js
import api from './api';

export const categoryService = {
  getCategories: () => api.get('/categories').then((r) => r.data),
  getCategoryById: (id) => api.get(`/categories/${id}`).then((r) => r.data),
  createCategory: (data) => api.post('/categories', data).then((r) => r.data),
  updateCategory: (id, data) => api.put(`/categories/${id}`, data).then((r) => r.data),
  deleteCategory: (id) => api.delete(`/categories/${id}`),
};
